provider.go

```diff
@@ -182,6 +166,20 @@ func (wp *WasmcloudProvider) OutgoingRpcClient(target string) *wrpcnats.Client {
 }
 
 func (wp *WasmcloudProvider) Start() error {
+	// update link events
+	for _, link := range wp.sourceLinks {
+		err := wp.putLink(link)
```

Comment:
Here, I believe calling `wp.putLink` will always no-op since the link will always be present, right? I think what we really want to do, since we already put the link in the provider's map, is call `wp.putSourceTargetLink` which is the developer specified callback, is that right?